src/leap/mail/incoming/service.py

```diff
@@ -458,10 +458,8 @@ def add_leap_header(ret):
                     self.LEAP_SIGNATURE_HEADER,
                     self.LEAP_SIGNATURE_INVALID)
             else:
-                decrmsg.add_header(
-                    self.LEAP_SIGNATURE_HEADER,
-                    self.LEAP_SIGNATURE_VALID,
-                    pubkey=signkey.fingerprint)
+                self._add_verified_signature_header(decrmsg,
+                                                    signkey.fingerprint)
```

Comment:
@meskio, actually `_do_fetch` returns a list of `SoledadDocuments` which seems contains encrypted message. For sure we could check the header on a decrypted message, but for now we cannot see how to get these decrypted messages after the `_do_fetch` call. Do you have any suggestion?

Here an example of `_do_fetch()` return: http://paste.debian.net/726745/